crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -71,6 +71,119 @@ impl Violation for Airflow3Removal {
     }
 }
 
+const REMOVED_CONTEXT_KEYS: [&str; 12] = [
+    "conf",
+    "execution_date",
+    "next_ds",
+    "next_ds_nodash",
+    "next_execution_date",
+    "prev_ds",
+    "prev_ds_nodash",
+    "prev_execution_date",
+    "prev_execution_date_success",
+    "tomorrow_ds",
+    "yesterday_ds",
+    "yesterday_ds_nodash",
+];
+
+fn extract_name_from_slice(slice: &Expr) -> Option<String> {
+    match slice {
+        Expr::StringLiteral(ExprStringLiteral { value, .. }) => Some(value.to_string()),
+        _ => None,
+    }
+}
+
+/// Checks for the use of deprecated Airflow context variables.
+///
+/// The function handles context keys accessed:
+/// - Directly using `context["key"]` or context.get("key").
+/// - Using `.get()` with variables derived from `get_current_context()`.
+/// - In custom operators, task decorators, and within templates or macros.
+///
+/// ## Examples
+///
+/// The following Python code would raise lint errors:
+///
+/// ```python
+/// @task
+/// def access_invalid_key_task(**context):
+///     print("access invalid key", context.get("execution_date"))  # Deprecated
+///
+/// @task
+/// def print_config(**context):
+///     execution_date = context["execution_date"]  # Deprecated
+///     prev_ds = context["prev_ds"]  # Deprecated
+///
+/// @task
+/// def print_config_from_context():
+///     context = get_current_context()
+///     execution_date = context["execution_date"]  # Deprecated
+/// ```
+/// In templates or custom plugins:
+///
+/// ```python
+/// task1 = DummyOperator(
+///     task_id="example_task",
+///     params={
+///         "execution_date": "{{ execution_date }}",  # Deprecated
+///     },
+/// )
+///
+/// class CustomMacrosPlugin(AirflowPlugin):
+///     name = "custom_macros"
+///     macros = {
+///         "execution_date_macro": lambda context: context["execution_date"],  # Deprecated
+///     }
+/// ```
+fn removed_context_variable(checker: &mut Checker, expr: &Expr) {
+    if let Expr::Subscript(ExprSubscript { value, slice, .. }) = expr {
```

Comment:
We can directly accept the `ExprSubscript` as the function argument if that's the only expression which is being checked here.

Reply:
changed it as per suggestion